tests/frameworks/OSXShims/src/LoggingNative.mm

```diff
@@ -15,44 +15,92 @@
 //******************************************************************************
 
 #import <LoggingNative.h>
+#import <Starboard/SmartTypes.h>
 #import <Foundation/NSString.h>
 
 #import <cstdarg>
 
+// NOTE: this function is assumed to only run in Little Endian environments like OSX
+static NSStringEncoding _getWcharEncoding() {
+    switch (sizeof(wchar_t)) {
+        case 2:
+            return NSUTF16LittleEndianStringEncoding;
+        case 4:
+            return NSUTF32LittleEndianStringEncoding;
+        default:
+            assert(false);
+    }
+}
+
+static NSStringEncoding getWcharEncoding() {
+    static NSStringEncoding encoding = _getWcharEncoding();
+    return encoding;
+}
+
 void TraceVerbose(const wchar_t* tag, const wchar_t* format, ...) {
-    NSString* formatString = [NSString stringWithFormat:@"[%ls] %ls", tag, format];
+    StrongId<NSString> nsTag;
```

Comment:
This is a file that will get compiled into all cross plat test binaries if I understand the build you set up ... some of those may have ARC on. This suggests we should use a mechanism that easily handles arc vs not which strong id does. Additionally it will be the same length as the alternative requires release statements in place of the attaches. The static cast is meh to me. I wouldn't mind a .get() method on a strong id so its more concise. 

---
In reply to: [74295731](https://github.com/Microsoft/WinObjC/pull/740#discussion_r74295731) [](ancestors = 74295731,74170322,74169017)